### PR TITLE
debian: add a zenity|kdialog suggests

### PR DESCRIPTION
**Pull request by mvo5**
The `snap userd` code makes use of either zenity or kdialog to
ask questions if xdg-setting defaults change. We should suggest
them. I inially added a recommend but this will pull in too
many extra packages on a server install so a recommends seems
to be more appropriate.

This PR also fixes a bug in our prepare script - if extra
packages get installed when building the image and those
extra package create users (like e.g. colord) then our
extraction for the "test" user does no longer work. This PR
fixes that extraction too.

### CODE REVIEW

**Comment by Conan-Kudo** [comment]
@mvo5 Should we be having zenity or kdialog installed everywhere? What's driving this change?

File: tests/lib/prepare.sh [hunk 1]
```diff
@@ -377,9 +377,9 @@ EOF
             # the passwd from core without root
             tail -n +2 "$UNPACKD/etc/$f" > /mnt/system-data/var/lib/extrausers/$f
             # append this systems root user so that linode can connect
-            head -n1 /etc/$f >> /mnt/system-data/var/lib/extrausers/$f
+            grep "^root:" /etc/$f >> /mnt/system-data/var/lib/extrausers/$f
```
Comment (zyga) [inline tests/lib/prepare.sh]: Hmm? Can you explain this please?

**Comment by mvo5** [comment]
@zyga I'm not sure if the rpm/arch world have a concept of a weak-weak dependency like suggests. As for 14.04, sure, let me add it.

**Comment by Conan-Kudo** [comment]
@mvo5 We have it in RPM:

* http://rpm.org/user_doc/dependencies.html#weak-dependencies
* http://rpm.org/user_doc/boolean_dependencies.html

We're using it in the Fedora spec now: https://github.com/snapcore/snapd/blob/master/packaging/fedora/snapd.spec#L100-L108

**Comment by bboozzoo** [comment]
For Arch there's [`optdepends`](https://wiki.archlinux.org/index.php/PKGBUILD#optdepends) although it's not as smart as in debian/rpm. An example is right here: https://github.com/snapcore/snapd/blob/master/packaging/arch/PKGBUILD#L23

**Comment by zyga** [comment]
I really love how various communities can come together around snaps. Thank you everyone!

**Comment by bboozzoo** [comment]
Needs deconflicting.

**Comment by bboozzoo** [comment]
@zyga you have requested changes in this PR, can you take another look?

FWIW. we could also add `xmessage` support if neither `zenity` nor `kdialog` are available.

**Pull request merged**